### PR TITLE
Bug/tr 1909/incorect answers represented in review

### PR DESCRIPTION
**Pull request by poyuki**
# [TR-1909](https://oat-sa.atlassian.net/browse/TR-1909)
depend on https://github.com/oat-sa/extension-tao-outcomeui/pull/399
## Summary
about duplicated displaying of responses for the same items in lti review mode
## How to reproduce

* execute delivery of test with duplicated items ([exmple](https://github.com/oat-sa/extension-lti-test-review/files/7603419/qa_check_1637742584.zip))
* run lti review mode for execution
* response of the last duplicated item will be provided for all duplicated items answers

https://user-images.githubusercontent.com/37804293/143454423-c068fa80-c6b2-40c7-ada0-625e33756b60.mp4

## How to test fix
* install extension with pr branch 
* execute steps from `How  to reproduce`
* right response items will be displayed

https://user-images.githubusercontent.com/37804293/143454328-a764e167-057a-4fbc-9adb-4d4f71829632.mp4

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -28,7 +28,8 @@
     "oat-sa/extension-tao-proctoring" : ">=20.0.0",
     "oat-sa/extension-tao-testqti" : ">=41.0.0",
     "oat-sa/extension-tao-outcome" : ">=13.0.0",
-    "oat-sa/extension-tao-testqti-previewer" : ">=3.0.0"
+    "oat-sa/extension-tao-testqti-previewer" : ">=3.0.0",
+    "oat-sa/extension-tao-outcomeui": "dev-bug/TR-1909/incorect-answers-represented-in-review as 100.0.0"
```
Comment (wazelin) [inline composer.json:32]: Bumping a major version of a dependency is a breaking change too.
IMO, extension-lti-test-review's major version should be bumped as well.

**Review by wazelin — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (poyuki) [reply composer.json:32]: this pr already has breaking change commit, also this comment  (https://github.com/oat-sa/extension-lti-test-review/pull/94#issuecomment-979187680), as i understand, means that major version will be bumped by this pr

File: models/QtiRunnerInitDataBuilder.php [hunk 2]
```diff
@@ -228,21 +229,25 @@ private function fillItemsData($itemId, $itemRef, $itemData)
         ];
     }
 
-    protected function getResultVariables($resultId, $filterSubmission, $filterTypes = array())
+    /**
+     * @throws common_exception_Error
+     * @throws common_Exception
+     */
+    protected function getResultVariables($resultId, $filterSubmission, $filterTypes = []): array
```
Comment (quintanilhar) [inline models/QtiRunnerInitDataBuilder.php:236]: That would be a great moment to create a test for this class, at least to cover these changes to ensure we won't have the same issue again.

**Review by quintanilhar — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [X] New code is respecting code style rules
- [X] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful
- [X] Pull request's target is not `master`
- [X] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [X] Commits messages are meaningful
- [X] Commits are atomic
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Review by wazelin — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`

**Pull request merged**